dm/master/election.go

```diff
@@ -38,11 +38,15 @@ func (s *Server) electionNotify(ctx context.Context) {
 			if leaderInfo == nil {
 				if s.leader == oneselfLeader {
 					s.scheduler.Close()
+					s.Lock()
 					s.leader = ""
+					s.Unlock()
 
 					log.L().Info("current member retire from the leader", zap.String("current member", s.cfg.Name))
 				}
+				s.Lock()
 				s.closeLeaderClient()
+				s.Unlock()
```

Comment:
How about putting these two `Lock` into only one to ensure the state is consistent?

Reply:
update in 0a352a3